### PR TITLE
Residual and add_embeding fix

### PR DESCRIPTION
**Pull request by ciebson123**
# Description
Blanks in the same sequence accumulate changes to embedding and encoder_output from other blanks (previously, they just added it to preblank)

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [ ] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [ ] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

**Comment by ciebson123** [comment]
Looks like it's working: https://app.neptune.ai/o/pmtest/org/llm-random/runs/compare?viewId=9a15aa36-1063-49c4-807d-53393abb118a&dash=charts&query=&sortBy=%5B%22sys%2Fcreation_time%22%5D&sortFieldType=%5B%22datetime%22%5D&sortFieldAggregationMode=%5B%22auto%22%5D&sortDirection=%5B%22descending%22%5D&lbViewUnpacked=true&compare=MwFgrGA0CMkOxA

**Comment by maciek-pioro** [comment]
It's probably alright, but we'd like to focus on fixing the residual blank, not adding to embedding.

File: research/blanks/model.py [hunk 1]
```diff
@@ -194,12 +197,17 @@ def forward(self, encoder_output: torch.Tensor, model_input: torch.Tensor):
                 )
 
             for _ in range(self.n_blanks):
-                preblank_encoder_output = shift_right(preblank_encoder_output)
-                encoder_output.add_(preblank_encoder_output * abs(self.preblank_weight))
+                current_positions = shift_right(current_positions)
+                encoder_accumulator = shift_right(encoder_accumulator)
+                encoder_output.add_(encoder_accumulator)
+                encoder_accumulator = encoder_output * current_positions
+
```
Comment (maciek-pioro) [inline research/blanks/model.py:204]: We are missing preblank weight here

Comment (ciebson123) [reply research/blanks/model.py:204]: I don't think so. It first adds an accumulated change (with preblank) to output and then basically copies whatever the result was into the accumulator

Comment (maciek-pioro) [reply research/blanks/model.py:204]: Yeah, but we don't even use the `preblank_weight` variable

File: research/blanks/model.py [hunk 2]
```diff
@@ -171,7 +171,10 @@ def forward(self, encoder_output: torch.Tensor, model_input: torch.Tensor):
         is_blank = model_input.eq(self.blank_token_id)
         is_first_blank = get_first_blanks_in_series(is_blank)
         is_preblank = shift_left(is_first_blank)
-        preblank_encoder_output = encoder_output * is_preblank.unsqueeze(-1)
+
+        current_positions = is_preblank.unsqueeze(-1)
```
Comment (maciek-pioro) [inline research/blanks/model.py:175]: Maybe `current_accumulator_positions`?

File: research/blanks/model.py (hunk 1 above)
Comment (ciebson123) [reply research/blanks/model.py:204]: Ah, I see, sorry, didn't see that before

**Review by maciek-pioro — APPROVED** [review]

**Pull request merged**